crates/sui-core/src/authority_server.rs

```diff
@@ -611,6 +611,11 @@ impl ValidatorService {
                 ))
                 .into()
             );
+
+            self.state.check_system_overload_for_ping(
```

Comment:
Can we call this later, maybe before iterating over the txns when this is not a Ping request?